README_python.md

```diff
@@ -0,0 +1,273 @@
+    Input<Buffer<uint8_t, 2>> input{"input"};
+    Input<uint8_t> mask{"mask"};
+    Output<Buffer<uint8_t, 2>> output{"output"};
+
+    void generate() {
+        output(x, y) = input(x, y) & mask;
+        output.vectorize(x, 8).compute_root();
+    }
+};
+HALIDE_REGISTER_GENERATOR(MyFilter, my_filter)
+```
+
+If you are using CMake, the simplest thing is to use
+`add_python_aot_extension()`, defined in PythonExtensionHelpers.cmake:
+
+```
+add_python_aot_extension(my_filter
+                         GENERATOR my_filter_generator
+                         SOURCES my_filter_generator.cpp
+                         [ FEATURES ... ]
+                         [ PARAMS ... ])
+```
+
+This compiles the Generator code in `my_filter_generator.cpp` with the
+registered name `my_filter` to produce the target `my_filter`, which is a Python
+Extension in the form of a shared library (e.g.,
+`foo.cpython-310-x86_64-linux-gnu.so`).
+
+### Calling a C++ Generator from Python
+
+As long as the shared library is in `PYTHONPATH`, it can be imported and used
+directly. For the example above:
+
+```
+from my_filter import my_filter
+import imageio
+import numpy as np
+
+# Read in some file for input
+input_buf = imageio.imread("/path/to/some/file.png")
+assert input_buf.dtype == np.uint8
+
+# arbitrary
+mask = 0x7f
+
+# create a Buffer-compatible object for the output; we'll use np.array
+output_buf = np.empty(input_buf.shape, dtype=input_buf.dtype)
+
+# Note, Python code throws exception for error conditions rather than returning an int
+my_filter(input_buf, mask, output_buf)
+
+imageio.imsave("/tmp/masked_file.png", output_buf)
+```
+
+Above, we're using common Python utilities (`numpy`, `imageio`) to construct the
+input/output buffers we want to pass to Halide.
+
+**Note**: Getting the memory order correct can be a little confusing for numpy.
+By default numpy uses "C-style"
+[row-major](https://docs.scipy.org/doc/numpy-1.13.0/reference/internals.html)
+order, which sounds like the right option for Halide; however, this nomenclature
+assumes the matrix-math convention of ordering axes as `[rows, cols]`, whereas
+Halide (and imaging code in general) generally assumes `[x, y]` (i.e., `[cols,
+rows]`). Thus what you usually want in Halide is column-major ordering. This
+means numpy arrays, by default, come with the wrong memory layout for Halide.
+But if you construct the numpy arrays yourself (like above), you can pass
+`order='F'` to make numpy use the Halide-compatible memory layout. If you're
+passing in an array constructed somewhere else, the easiest thing to do is to
+`.transpose()` it before passing it to your Halide code.)
+
+## Keeping Up To Date
+
+If you use the Halide Bindings for Python inside Google, you are *strongly*
+encouraged to
+[subscribe to announcements for new releases of Halide](https://github.blog/changelog/2018-11-27-watch-releases/),
+as it is likely that enhancements and tweaks to our Python support will be made
+in future releases.
```

Comment:
Is this still relevant?

Reply:
Definitely, as I suspect that we'll find missing pieces & sharp corners that we'll want to fix sooner rather than later